<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />


<meta name="author" content="Mingxuan Cai" />

<meta name="date" content="2024-04-30" />

<title>LDL example with 1000 Genomes Reference Panel</title>

<script src="site_libs/header-attrs-2.16/header-attrs.js"></script>
<script src="site_libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/cosmo.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<style>h1 {font-size: 34px;}
       h1.title {font-size: 38px;}
       h2 {font-size: 30px;}
       h3 {font-size: 24px;}
       h4 {font-size: 18px;}
       h5 {font-size: 16px;}
       h6 {font-size: 12px;}
       code {color: inherit; background-color: rgba(0, 0, 0, 0.04);}
       pre:not([class]) { background-color: white }</style>
<script src="site_libs/jqueryui-1.11.4/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<link href="site_libs/highlightjs-9.12.0/textmate.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>
<link href="site_libs/font-awesome-5.1.0/css/all.css" rel="stylesheet" />
<link href="site_libs/font-awesome-5.1.0/css/v4-shims.css" rel="stylesheet" />

<link rel="icon" href="https://github.com/workflowr/workflowr-assets/raw/main/img/reproducible.png">
<!-- Add a small amount of space between sections. -->
<style type="text/css">
div.section {
  padding-top: 12px;
}
</style>



<style type="text/css">
  code{white-space: pre-wrap;}
  span.smallcaps{font-variant: small-caps;}
  span.underline{text-decoration: underline;}
  div.column{display: inline-block; vertical-align: top; width: 50%;}
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
  ul.task-list{list-style: none;}
    </style>

<style type="text/css">code{white-space: pre;}</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>









<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
details > summary > p:only-child {
  display: inline;
}
pre code {
  padding: 0;
}
</style>


<style type="text/css">
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #adb5bd;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script type="text/javascript">
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark the anchor link active (and if it's in a dropdown, also mark that active)
  var dropdown = menuAnchor.closest('li.dropdown');
  if (window.bootstrap) { // Bootstrap 4+
    menuAnchor.addClass('active');
    dropdown.find('> .dropdown-toggle').addClass('active');
  } else { // Bootstrap 3
    menuAnchor.parent().addClass('active');
    dropdown.addClass('active');
  }

  // Navbar adjustments
  var navHeight = $(".navbar").first().height() + 15;
  var style = document.createElement('style');
  var pt = "padding-top: " + navHeight + "px; ";
  var mt = "margin-top: -" + navHeight + "px; ";
  var css = "";
  // offset scroll position for anchor links (for fixed navbar)
  for (var i = 1; i <= 6; i++) {
    css += ".section h" + i + "{ " + pt + mt + "}\n";
  }
  style.innerHTML = "body {" + pt + "padding-bottom: 40px; }\n" + css;
  document.head.appendChild(style);
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "&#xe258;";
  border: none;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
  background-color: transparent;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->



<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}

@media print {
.toc-content {
  /* see https://github.com/w3c/csswg-drafts/issues/4434 */
  float: right;
}
}

.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>



</head>

<body>


<div class="container-fluid main-container">


<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-bs-toggle="collapse" data-target="#navbar" data-bs-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">XMAP</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Home</a>
</li>
<li>
  <a href="about.html">Installation</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">
    Tutorial
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
    <li>
      <a href="quick_start.html">Quick Start</a>
    </li>
    <li>
      <a href="confounding.html">Simulated example with confounding effects</a>
    </li>
    <li>
      <a href="LDL.html">Real data example with LDL GWAS and UK Biobank reference genotypes</a>
    </li>
    <li>
      <a href="LDL_TGP.html">Real data example with LDL GWAS and 1000 Genomes reference genotypes</a>
    </li>
  </ul>
</li>
<li>
  <a href="license.html">License</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        <li>
  <a href="https://github.com/YangLabHKUST/XMAP">
    <span class="fab fa-github"></span>
     
    Source code
  </a>
</li>
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div id="header">



<h1 class="title toc-ignore">LDL example with 1000 Genomes Reference
Panel</h1>
<h4 class="author">Mingxuan Cai</h4>
<h4 class="date">2024-04-30</h4>

</div>


<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-report" data-toggle="collapse" data-target="#workflowr-report">
<span class="glyphicon glyphicon-list" aria-hidden="true"></span>
workflowr <span class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span>
</button>
</p>
<div id="workflowr-report" class="collapse">
<ul class="nav nav-tabs">
<li class="active">
<a data-toggle="tab" href="#summary">Summary</a>
</li>
<li>
<a data-toggle="tab" href="#checks"> Checks <span
class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span> </a>
</li>
<li>
<a data-toggle="tab" href="#versions">Past versions</a>
</li>
</ul>
<div class="tab-content">
<div id="summary" class="tab-pane fade in active">
<p>
<strong>Last updated:</strong> 2024-04-30
</p>
<p>
<strong>Checks:</strong> <span
class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> 6
<span class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span> 1
</p>
<p>
<strong>Knit directory:</strong> <code>XMAP-tutorial/</code> <span
class="glyphicon glyphicon-question-sign" aria-hidden="true"
title="This is the local directory in which the code in this file was executed.">
</span>
</p>
<p>
This reproducible <a href="https://rmarkdown.rstudio.com">R Markdown</a>
analysis was created with <a
  href="https://github.com/workflowr/workflowr">workflowr</a> (version
1.7.0). The <em>Checks</em> tab describes the reproducibility checks
that were applied when the results were created. The <em>Past
versions</em> tab lists the development history.
</p>
<hr>
</div>
<div id="checks" class="tab-pane fade">
<div id="workflowr-checks" class="panel-group">
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRMarkdownfilestronguncommittedchanges">
<span class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span> <strong>R Markdown file:</strong> uncommitted
changes </a>
</p>
</div>
<div id="strongRMarkdownfilestronguncommittedchanges"
class="panel-collapse collapse">
<div class="panel-body">
<p>The R Markdown is untracked by Git. To know which version of the R
Markdown file created these results, you’ll want to first commit it to
the Git repo. If you’re still working on the analysis, you can ignore
this warning. When you’re finished, you can run
<code>wflow_publish</code> to commit the R Markdown file and build the
HTML.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongEnvironmentstrongempty">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Environment:</strong> empty </a>
</p>
</div>
<div id="strongEnvironmentstrongempty" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! The global environment was empty. Objects defined in the
global environment can affect the analysis in your R Markdown file in
unknown ways. For reproduciblity it’s best to always run the code in an
empty environment.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSeedstrongcodesetseed20230213code">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Seed:</strong>
<code>set.seed(20230213)</code> </a>
</p>
</div>
<div id="strongSeedstrongcodesetseed20230213code"
class="panel-collapse collapse">
<div class="panel-body">
<p>The command <code>set.seed(20230213)</code> was run prior to running
the code in the R Markdown file. Setting a seed ensures that any results
that rely on randomness, e.g. subsampling or permutations, are
reproducible.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSessioninformationstrongrecorded">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Session information:</strong>
recorded </a>
</p>
</div>
<div id="strongSessioninformationstrongrecorded"
class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Recording the operating system, R version, and package
versions is critical for reproducibility.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongCachestrongnone">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Cache:</strong> none </a>
</p>
</div>
<div id="strongCachestrongnone" class="panel-collapse collapse">
<div class="panel-body">
<p>Nice! There were no cached chunks for this analysis, so you can be
confident that you successfully produced the results during this
run.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongFilepathsstrongrelative">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>File paths:</strong> relative </a>
</p>
</div>
<div id="strongFilepathsstrongrelative" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Using relative paths to the files within your workflowr
project makes it easier to run your code on other machines.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRepositoryversionstrongahrefhttpsgithubcommxcaiXMAPtutorialtree36aec6276bc48c656316eebb907c347f44473aa5targetblank36aec62a">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Repository version:</strong>
<a href="https://github.com/mxcai/XMAP-tutorial/tree/36aec6276bc48c656316eebb907c347f44473aa5" target="_blank">36aec62</a>
</a>
</p>
</div>
<div
id="strongRepositoryversionstrongahrefhttpsgithubcommxcaiXMAPtutorialtree36aec6276bc48c656316eebb907c347f44473aa5targetblank36aec62a"
class="panel-collapse collapse">
<div class="panel-body">
<p>
Great! You are using Git for version control. Tracking code development
and connecting the code version to the results is critical for
reproducibility.
</p>
<p>
The results in this page were generated with repository version
<a href="https://github.com/mxcai/XMAP-tutorial/tree/36aec6276bc48c656316eebb907c347f44473aa5" target="_blank">36aec62</a>.
See the <em>Past versions</em> tab to see a history of the changes made
to the R Markdown and HTML files.
</p>
<p>
Note that you need to be careful to ensure that all relevant files for
the analysis have been committed to Git prior to generating the results
(you can use <code>wflow_publish</code> or
<code>wflow_git_commit</code>). workflowr only checks the R Markdown
file, but you know if there are other scripts or data files that it
depends on. Below is the status of the Git repository when the results
were generated:
</p>
<pre><code>
Ignored files:
    Ignored:    .DS_Store
    Ignored:    .Rhistory
    Ignored:    .Rproj.user/
    Ignored:    data/.DS_Store

Untracked files:
    Untracked:  analysis/LDL_TGP.Rmd
    Untracked:  data/xmap_susie_LDL_TGP.RData

Unstaged changes:
    Modified:   analysis/LDL.Rmd
    Modified:   analysis/_site.yml

</code></pre>
<p>
Note that any generated files, e.g. HTML, png, CSS, etc., are not
included in this status report because it is ok for generated content to
have uncommitted changes.
</p>
</div>
</div>
</div>
</div>
<hr>
</div>
<div id="versions" class="tab-pane fade">
<p>
There are no past versions. Publish this analysis with
<code>wflow_publish()</code> to start tracking its development.
</p>
<hr>
</div>
</div>
</div>
<p>We demonstrate how to fit XMAP with reference genotypes from publicly
available reference genotypes. For illustration, we consider the LDL
GWAS summary statistics from the <a
href="https://mxcai.github.io/XMAP-tutorial/LDL.html">full example</a>,
while using the 1000 Genomes data to construct EUR and AFR LD matrices.
In the following, we introduce how to construct LD matrices from the
1000 Genomes data and fit XMAP with the constructed LD matrices step by
step.</p>
<div id="step-1-download-and-preprocess-1000-genomes-genotypes"
class="section level2">
<h2>Step 1: Download and preprocess 1000 Genomes genotypes</h2>
<p>We first obtain GRCH37 1000 Genomes genotype files from <a
href="https://www.cog-genomics.org/plink/2.0/resources">plink</a> and
convert them to plink1 format.</p>
<pre><code>plink2 \\
  --autosome \\
  --make-bed \\
  --max-alleles 2 \\
  --out all_phase3_noQC \\
  --pfile all_phase3 \\
  --rm-dup exclude-all \\
  --snps-only just-acgt \\
  --var-filter</code></pre>
<p>Next, we perform QC, extract EUR and AFR samples, and obtain the
genotype files for the target chromosome (chromosome 8 here).</p>
<pre><code>plink2 \\
  --bfile all_phase3_noQC \\
  --chr 8 \\
  --hwe 1e-10 \\
  --keep iid_EUR.txt \\
  --maf 0.005 \\
  --make-bed \\
  --out all_phase3_EUR_chr8 \\
  --var-filter
  
plink2 \\
  --bfile all_phase3_noQC \\
  --chr 8 \\
  --hwe 1e-10 \\
  --keep iid_AFR.txt \\
  --maf 0.005 \\
  --make-bed \\
  --out all_phase3_AFR_chr8 \\
  --var-filter</code></pre>
</div>
<div
id="step-2-construct-ld-matrices-for-eur-and-afr-in-the-target-region"
class="section level2">
<h2>Step 2: Construct LD matrices for EUR and AFR in the target
region</h2>
<p>Once the plink files are ready, we can read summary statistics and
extract SNPs from the summary data and reference genotype in the target
region. Here, we focus on 20000001-23000001 in chromosome 8, which is
present in our manuscript.</p>
<pre class="r"><code>library(XMAP)
library(susieR)
library(data.table)
library(Matrix)
library(snpStats)

# read sumstats
sumstat_EUR &lt;- fread(&quot;/Users/cmx/Documents/Research/Project/Fine_Mapping/sumstats/LDL_allSNPs_UKBNealLab_summary_format.txt&quot;)
sumstat_AFR &lt;- fread(&quot;/Users/cmx/Documents/Research/Project/Fine_Mapping/sumstats/LDL_AFR_GLGC_summary_format.txt&quot;)

# read genotype information
bim_EUR &lt;- fread(&quot;/Users/cmx/Documents/Research/Data/GWAS/Genotype/all_phase3_EUR_chr8.bim&quot;)
bim_AFR &lt;- fread(&quot;/Users/cmx/Documents/Research/Data/GWAS/Genotype/all_phase3_AFR_chr8.bim&quot;)

# detect allele ambiguous SNPs
idx_amb_EUR &lt;- (bim_EUR$V5 == comple(bim_EUR$V6))
idx_amb_AFR &lt;- (bim_AFR$V5 == comple(bim_AFR$V6))

# identify overlapping SNPs on region chr8_20000001_23000001
snps &lt;- Reduce(intersect, list(bim_AFR$V2[!idx_amb_AFR &amp; bim_AFR$V4 &gt; 20000001 &amp; bim_AFR$V4 &lt; 23000001], bim_EUR$V2[!idx_amb_EUR], sumstat_AFR$SNP, sumstat_EUR$SNP))

# read genotype matrices
geno_EUR &lt;- read.plink(&quot;/Users/cmx/Documents/Research/Data/GWAS/Genotype/all_phase3_EUR_chr8.bed&quot;)
X_EUR &lt;- as(geno_EUR$genotypes, &quot;numeric&quot;)
X_EUR_i &lt;- X_EUR[,match(snps, bim_EUR$V2)]

geno_AFR &lt;- read.plink(&quot;/Users/cmx/Documents/Research/Data/GWAS/Genotype/all_phase3_AFR_chr8.bed&quot;)
X_AFR &lt;- as(geno_AFR$genotypes, &quot;numeric&quot;)
X_AFR_i &lt;- X_AFR[,match(snps, bim_AFR$V2)]

bim_EUR_i &lt;- bim_EUR[match(snps, bim_EUR$V2), ]
bim_AFR_i &lt;- bim_AFR[match(snps, bim_AFR$V2), ]

# align genotype alleles using EUR as reference
idx_flip &lt;- which(bim_AFR_i$V5!=bim_EUR_i$V5 &amp; bim_AFR_i$V5!=comple(bim_EUR_i$V5))
# flip allels in AFR genotypes
X_AFR_i[,idx_flip] &lt;- 2 - X_AFR_i[,idx_flip]

# compute LD matrices
R_EUR &lt;- cor(X_EUR_i)
R_AFR &lt;- cor(X_AFR_i)</code></pre>
<p>The <code>R_EUR</code> and <code>R_AFR</code> constructed above will
be used for fitting XMAP in the subsequent analysis. We also extract
SNPs from summary data for the target region and flip alleles in the
summary data to match the reference genotype.</p>
<pre class="r"><code># extract target region snps from sumstats data
sumstat_AFR_i &lt;- sumstat_AFR[match(snps, sumstat_AFR$SNP), ]
sumstat_EUR_i &lt;- sumstat_EUR[match(snps, sumstat_EUR$SNP), ]

# flip alleles in sumstats
z_afr &lt;- sumstat_AFR_i$beta / sumstat_AFR_i$se
z_eur &lt;- sumstat_EUR_i$Z

idx_flip &lt;- which(sumstat_AFR_i$A1 != bim_EUR_i$V5 &amp; sumstat_AFR_i$A1 != comple(bim_EUR_i$V5))
z_afr[idx_flip] &lt;- -z_afr[idx_flip]

idx_flip &lt;- which(sumstat_EUR_i$A1 != bim_EUR_i$V5 &amp; sumstat_EUR_i$A1 != comple(bim_EUR_i$V5))
z_eur[idx_flip] &lt;- -z_eur[idx_flip]</code></pre>
</div>
<div id="step-3-fit-xmap-with-the-constructed-ld-matrices"
class="section level2">
<h2>Step 3: Fit XMAP with the constructed LD matrices</h2>
<p>With the LD matrices <code>R_EUR</code> and <code>R_AFR</code>, and
the summary statistics <code>z_eur</code> and <code>z_afr</code>, we can
fit XMAP to obtain the causal SNPs and credible sets. Here, we set the
number of causal SNPs to 10. We set the <code>C1</code> and
<code>C2</code> values to the estimated LDSC intercepts. Details can be
found at <a href="https://mxcai.github.io/XMAP-tutorial/LDL.html">the
step 1 in the full example</a>.</p>
<pre class="r"><code># LDSC intercepts
c1 &lt;- 1.066501
c2 &lt;- 1.095526

# fit XMAP
xmap &lt;- XMAP(simplify2array(list(R_EUR, R_AFR)), cbind(z_eur, z_afr), n=c(median(sumstat_EUR_i$N), median(sumstat_AFR_i$N)),
             K = 10, Omega = OmegaHat, Sig_E = c(c1, c2), tol = 1e-6,
             maxIter = 200, estimate_residual_variance = F, estimate_prior_variance = T,
             estimate_background_variance = F)
cs1 &lt;- get_CS(xmap, Xcorr = R_AFR, coverage = 0.9, min_abs_corr = 0.1)
cs2 &lt;- get_CS(xmap, Xcorr = R_EUR, coverage = 0.9, min_abs_corr = 0.1)
cs_xmap &lt;- cs1$cs[intersect(names(cs1$cs), names(cs2$cs))]
pip_xmap &lt;- get_pip(xmap$gamma)
plot_CS(pip_xmap, cs_xmap, main = &quot;XMAP&quot;,b = (bim_EUR_i$V2 == &quot;rs900776&quot;))</code></pre>
<p><img src="figure/LDL_TGP.Rmd/unnamed-chunk-4-1.png" width="672" style="display: block; margin: auto;" />
For comparison, we also fit SuSiE with AFR and EUR data separately and
plot the credible sets.</p>
<pre class="r"><code>susie_EUR &lt;- susie_rss(z=z_eur,R=R_EUR,n=median(sumstat_EUR_i$N))
susie_plot(susie_EUR,y=&quot;PIP&quot;,b = (bim_EUR_i$V2 == &quot;rs900776&quot;))

susie_AFR &lt;- susie_rss(z=z_afr,R=R_AFR,n=median(sumstat_AFR_i$N))
susie_plot(susie_AFR,y=&quot;PIP&quot;,b = (bim_EUR_i$V2 == &quot;rs900776&quot;))</code></pre>
<p><img src="figure/LDL_TGP.Rmd/unnamed-chunk-6-1.png" width="672" style="display: block; margin: auto;" /><img src="figure/LDL_TGP.Rmd/unnamed-chunk-6-2.png" width="672" style="display: block; margin: auto;" /></p>
</div>
<div id="caution-on-out-sample-ld-reference" class="section level2">
<h2>Caution on out-sample LD reference</h2>
<p>In this example, we used publicly available data from 1000 Genomes
Project to construct the LD matrices for EUR and AFR populations.
However, the LD matrices constructed from the reference data may produce
unreliable results in fine-mapping. <strong>In practice, we recommend
using the in-sample genotype data to construct the LD matrices for the
target population. One can follow the above steps to construct their own
LD matrices and fit XMAP with the in-sample genotype data.</strong></p>
<br>
<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-sessioninfo" data-toggle="collapse" data-target="#workflowr-sessioninfo" style="display: block;">
<span class="glyphicon glyphicon-wrench" aria-hidden="true"></span>
Session information
</button>
</p>
<div id="workflowr-sessioninfo" class="collapse">
<pre class="r"><code>sessionInfo()</code></pre>
<pre><code>R version 4.1.2 (2021-11-01)
Platform: aarch64-apple-darwin20 (64-bit)
Running under: macOS 13.6.3

Matrix products: default
BLAS:   /opt/homebrew/Cellar/openblas/0.3.19/lib/libopenblasp-r0.3.19.dylib
LAPACK: /Library/Frameworks/R.framework/Versions/4.1-arm64/Resources/lib/libRlapack.dylib

locale:
[1] en_US.UTF-8/en_US.UTF-8/en_US.UTF-8/C/en_US.UTF-8/en_US.UTF-8

attached base packages:
[1] stats     graphics  grDevices utils     datasets  methods   base     

other attached packages:
[1] susieR_0.12.35 XMAP_1.0      

loaded via a namespace (and not attached):
 [1] tidyselect_1.2.0   xfun_0.32          bslib_0.4.0        lattice_0.20-45   
 [5] colorspace_2.0-2   vctrs_0.6.3        generics_0.1.1     htmltools_0.5.4   
 [9] yaml_2.3.5         utf8_1.2.2         rlang_1.1.1        mixsqp_0.3-43     
[13] jquerylib_0.1.4    later_1.3.0        pillar_1.9.0       glue_1.6.2        
[17] matrixStats_0.61.0 lifecycle_1.0.3    plyr_1.8.6         stringr_1.4.1     
[21] munsell_0.5.0      gtable_0.3.0       workflowr_1.7.0    evaluate_0.16     
[25] knitr_1.40         fastmap_1.1.0      httpuv_1.6.5       irlba_2.3.5       
[29] fansi_0.5.0        highr_0.9          Rcpp_1.0.10        promises_1.2.0.1  
[33] scales_1.1.1       cachem_1.0.6       jsonlite_1.8.0     fs_1.5.2          
[37] ggplot2_3.3.5      digest_0.6.29      stringi_1.7.8      dplyr_1.1.2       
[41] rprojroot_2.0.2    grid_4.1.2         cli_3.6.1          tools_4.1.2       
[45] magrittr_2.0.3     sass_0.4.2         tibble_3.2.1       crayon_1.5.1      
[49] pkgconfig_2.0.3    Matrix_1.5-4.1     rmarkdown_2.16     reshape_0.8.8     
[53] rstudioapi_0.13    R6_2.5.1           git2r_0.31.0       compiler_4.1.2    </code></pre>
</div>
</div>


<!-- Adjust MathJax settings so that all math formulae are shown using
TeX fonts only; see
https://docs.mathjax.org/en/latest/web/configuration.html. This will make
the presentation more consistent at the cost of the webpage sometimes
taking slightly longer to load. Note that this only works because the
footer is added to webpages before the MathJax javascript. -->
<script type="text/x-mathjax-config">
  MathJax.Hub.Config({
    "HTML-CSS": { availableFonts: ["TeX"] }
  });
</script>




</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.odd').parent('tbody').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open');
  });
});
</script>

<!-- code folding -->

<script>
$(document).ready(function ()  {

    // temporarily add toc-ignore selector to headers for the consistency with Pandoc
    $('.unlisted.unnumbered').addClass('toc-ignore')

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_');
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
